ArcSlider/ArcSliderBehaviorDecorator.js

```diff
@@ -0,0 +1,121 @@
+import hoc from '@enact/core/hoc';
+import platform from '@enact/core/platform';
+import PropTypes from 'prop-types';
+import React from 'react';
+
+import {positionToAngle} from '../Arc/utils';
+import {angleToValue} from './utils';
+
+// Adds agate-specific arcSlider behaviors
+const ArcSliderBehaviorDecorator = hoc((config, Wrapped) => {
+	return class extends React.Component {
+		static displayName = 'ArcSliderBehaviorDecorator';
+
+		static propTypes = {
+			endAngle: PropTypes.number,
+			max: PropTypes.number,
+			min: PropTypes.number,
+			radius: PropTypes.number,
+			startAngle: PropTypes.number,
+			step: PropTypes.number,
+			strokeWidth: PropTypes.number
+		};
+
+		constructor (props) {
+			super(props);
+
+			this.componentRef = React.createRef();
+
+			this.state = {
+				value: props.min
+			};
+		}
+
+		onMouseDown = (ev) => {
+			const componentRef = this.componentRef.current;
+
+			if (!platform.touchscreen) {
+				if (componentRef) {
+					componentRef.addEventListener('mousemove', this.calculateNewValue);
+					componentRef.addEventListener('mouseup', this.removeMouseListeners);
+				}
```

Comment:
If using `onDragStart`, `onDrag`, `onDragEnd` in `Touchable`, we don't need `addEventListener` and `removeEventListener`.